types/bandagedbd__bdapi/index.d.ts

```diff
@@ -12,473 +12,1387 @@ import * as _ from 'lodash';
+         *
+         * @param argument Anything that should be used to add classnames.
+         */
+        className(argument: object | object[]): string;
+
+        /**
+         * Returns a function, that, as long as it continues to be invoked, will not be triggered. The function will be
+         * called after it stops being called for N milliseconds.
+         *
+         * Adapted from the version by David Walsh (https://davidwalsh.name/javascript-debounce-function)
+         *
+         * @param executor The function to be called after N milliseconds of inactivity
+         * @param delay The delay, in milliseconds
+         */
+        debounce(executor: () => void, delay: number): () => void;
+
+        /**
+         * Takes a string of HTML and escapes it using the brower's own escaping mechanism.
+         *
+         * @param html HTML to be escaped
+         */
+        escapeHTML(html: string): string;
+
+        /**
+         * Deep extends an object with a set of other objects. Objects later in the list of extenders have priority,
+         * that is to say if one sets a key to be a primitive, it will be overwritten with the next one with the same
+         * key. If it is an object, and the keys match, the object is extended. This happens recursively.
+         *
+         * @param extendee Object to be extended
+         * @param extenders Objects to extend with
+         */
+        extend(extendee: object, extenders: object[]): object;
+
+        /**
+         * Finds a value, subobject, or array from a tree that matches a specific filter. This is a DFS.
+         *
+         * @param tree Tree that should be walked
+         * @param searchFilter Filter to check against each object and subobject
+         * @param options Additional options to customize the search
+         */
+        findInTree(tree: object, searchFilter: (node: object) => boolean, options: UtilsFindInTreeOptions): any;
+    }
+}
 
+declare namespace WebpackModule {
     /**
-     * Determines if the input is valid and parseable JSON.
-     * @param data Data to test.
-     * @returns True if the data is valid, false otherwise.
+     * A function to use to filter modules. It is given exports, module, and moduleID. Return true to signify match.
      */
-    function testJSON(data: string): boolean;
+    type WebpackFilter = (module: any) => boolean;
+
+    interface WebpackModuleSearchOptions {
+        /**
+         * A function to use to filter modules.
+         *
+         * @param object The module being filtered
+         */
+        filter: WebpackFilter;
+        /**
+         * Whether to return only the first matching module. Defaults to true.
+         */
+        first?: boolean;
+        /**
+         * Whether to return default export when matching the default export. Defaults to true.
+         */
+        defaultExport?: boolean;
+        /**
+         * 	Whether to execute the filter on webpack exports. Defaults to false.
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/bandagedbd__bdapi/index.d.ts;
  Line: 1285;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

Reply:
Replaced tabs with spaces.